### DISCUSSION TITLE
opentrack 2022.3.2

### DISCUSSION THREAD

**Discussion by sthalik** [opentrack/opentrack#1557] [Announcements]
### Fixes

- Fix tracker/wii not loading on some systems due to a build issue (reported by @GO63-samara, @oliverCJ and @proxlamus).

<hr /><em>This discussion was created from the release <a href='https://github.com/opentrack/opentrack/releases/tag/opentrack-2022.3.2'>opentrack 2022.3.2</a>.</em>

**Comment by proxlamus** [comment]
Beautiful.  Works perfect now!  Thank you!

**Reply by jamescant** [reply]
wont open ive tried every thing even youtube videos idk ik im sad 
i wont have open track on my b day

**Reply by proxlamus** [reply]
Explain more? You tried to open it and what happens next?     In the input drop down what are you selecting?  What are you selecting in output?
What Webcam are you using?

**Reply by GO63-samara** [reply]
Maybe Windows XP ? ;)

**Reply by sthalik** [reply]
Has nothing on Windows Bandera.

**Comment by NatsukiRem** [comment]
Does this version work with MFS2020 (version: 1.27.21.0)? In the prog title it writes "unknown game"

**Reply by sthalik** [reply]
Probably. Because so far nobody complained about it.

**Reply by NatsukiRem** [reply]
Oh. I'm an idiot. I had to launch opentrack before the game start. Maybe I'll help somebody who as stupid as me via the comment. Thanks for the program!

**Reply by sthalik** [reply]
Only the first time, as it has to write out some registry entries.

**Comment by sthalik** [comment]
For developers -- you can now browse the repo through the `cgit` interface which is available as <<https://ananke.misaki.pl/cgit/>>.

**Comment by gibsonfm** [comment]
There are another changes beyond that fix?

**Reply by sthalik** [reply]
No. Are there any unreported issues?

**Reply by gibsonfm** [reply]
No, thanks.

**Comment by Trodaiche** [comment]
Hi 
I have an image on my camera when I open the Windows Camera app.

Ive installed the app and left as default...the only settings I have changed are input to 'UDP over Network' 

I have confirmed port 4242 and updated to latest [Visual C++ Redistributable x64]

When I select 'Start', Im getting the message 'No Video' and no tracking occurs?

Any advice thanks

**Reply by sthalik** [reply]
Didn't you mean UDP *output*? You're only able to show a preview on the computer with a camera-based *input* (such as PT, Aruco or neuralnet).

**Comment by Trodaiche** [comment]
Ah OK now I'm confused... Trying to follow videos... So I select udp on the output so what gets selected on the input?

**Reply by sthalik** [reply]
You have a webcam and want to use it... So select:
> a camera-based *input* (such as PT, Aruco or neuralnet).

**Reply by Trodaiche** [reply]
Thanks unfortunately when I select any of these input options I get
Pt - failed to open camera
The other options provide no movement

Thanks

**Reply by sthalik** [reply]
Can you select the camera from the dropdown box in PT settings?

**Reply by Trodaiche** [reply]
yes the camera is selected by default

**Reply by sthalik** [reply]
Could this possibly be an antivirus issue?

**Reply by Trodaiche** [reply]
ive tried turning it all off. .ill try again

**Reply by Trodaiche** [reply]
to add...i have also allowed the camera to be detected under windows settings

**Reply by Trodaiche** [reply]
No luck tried turning off antivirus and firewall settings

**Reply by sthalik** [reply]
Just to be sure, you could try deleting the file `opentrack-video-ps3eye.dll` inside the `modules/` directory then trying again.

**Reply by Trodaiche** [reply]
![image](https://user-images.githubusercontent.com/59740392/198652815-c019b685-812f-4874-917f-434c6a6c21e4.png)

afraid same issue

**Reply by GO63-samara** [reply]
Also check that the Output must not be UDP: 4242.
It's better to use Output: freetrack.

**Reply by Trodaiche** [reply]
Thanks guys neuralnet tracker for the input and freetrack for the output got things working. Appreciated!

**Comment by allarallas** [comment]
Downloaded a setup file today with following results - 
![image](https://user-images.githubusercontent.com/1219719/198824550-760c3fd0-15c7-49ac-abb5-13148adcb9c2.png)

Any comments?

**Reply by GO63-samara** [reply]
I think it's Bitdefender swearing at the free installer. Try using the portable version: opentrack-2022.3.2-win32-portable.7z

**Reply by allarallas** [reply]
> I think it's Bitdefender swearing at the free installer. Try using the portable version: opentrack-2022.3.2-win32-portable.7z

No. With 7z file is the same threat present, however pointing to a concrete file - The file <extraction path>\install\modules\ps3eye-subprocess.exe is infected with Trojan.GenericKD.62958882 and was moved to quarantine.

Why is there so slow reaction from the maintainers? It's the second month since the initial report and still no comments, security alerts or new releases.

**Reply by sthalik** [reply]
Because there's no virus.

**Reply by okayflamed** [reply]
Why do you always say that it's not a virus without maybe finding out why this happens in the first place? There is no reason a program like this should be false-flagged as being a trojan of all possibilities.

**Reply by sthalik** [reply]
> Why do you always say that it's not a virus without maybe finding out why this happens in the first place?

Every `.dll` file inside the archive was either built by me from the sources inside the repo or provided inside one of the SDKs by a device vendor. The installer is produced by a legitimate version of Inno Setup. It's not really feasible for viruses to happen themselves into the archive that I release shortly after doing a build.

opentrack does plenty of things that trojans also do. It logs all the keystrokes pressed inside any application in order to provide hotkeys for centering and the like, and even receives mouse movement in order to support shortcuts on extended mouse buttons. Then it uses hand-built versions of Qt and opencv that aren't already whitelisted.

Antivirus software can't really understand what a program does (this is mathematically impossible) for all possible programs so there's always guesswork involved.

It's also not feasible to prove that a given computer absolutely doesn't have viruses. They can hide inside mainboard's and CPU's management engines, or even hard drives as was done by the NSA on multiple hard drive vendors. For what it's worth, I run sysinternals autoruns on the dev machine and install malwarebytes sometimes.

**Reply by GO63-samara** [reply]
It would be nice to copy this detailed answer to a frequently repeated question to `Q&A`.

**Reply by okayflamed** [reply]
I just tried the portable version and everything is fine. This crap scared the heck out of me man lmao. I guess Windows doesn't like plain .exe files? Who knows

**Comment by McKangi** [comment]
I also think it's a virus, a trojan.

![Képernyőkép 2022-11-06 210724](https://user-images.githubusercontent.com/5057143/200192570-744f2188-9bcc-47b5-aca0-885a006b4940.jpg)

**Comment by MrSnuffy1972** [comment]
I get the following error when starting tracking with the Intel RealSense Technology Input:

Tracking briefly starts and it IS tracking me.  After 3 or 4 seconds it crashes with the error below:

OMP: Error #136: Cannot create thread.
OMP: System error #8: Not enough memory resources are available to process this command.
OMP: Error #178: Function GetExitCodeThread() failed:
OMP: System error #6: The handle is invalid.

I have a RealSense SR300 that's worked perfectly in the past.
I've tried running the application as administrator with no change in behaviour.

**Comment by Jab46** [comment]
Hello, 
my system has detected a threat into this 2022.3.2.
Trojan.GenericKD.62958882.
This threat is not présent in the previous version. 2022.3.1

**Comment by sopryshko** [comment]
Hello,
I have a problem with a video stream frame position. Usually, my head is in the centre of the image. But after ~2-5 minutes of running the tracker the image shifts and overlaps like this:
![screenshot](https://user-images.githubusercontent.com/47949991/205412011-d3bd4091-dcfc-49bc-8e4d-ab3047a44033.png)

After that, I have to restart tracking because it doesn't work.

Here is another example of the frame shift (it's less critical so I still able to use the tracker):
![113127](https://user-images.githubusercontent.com/47949991/205413040-1b79d353-dd4b-4fc9-9020-6c64afd86b0a.png)

Let me know if you need more info to fix this issue.

**Reply by sthalik** [reply]
The reason for this issue is dropped USB bulk ISO frames. Hopefully 567cc1993617335b2b9b46b1597065e1b2bd422a can fix it. Try replacing this executable file in the `modules` subdirectory.

[ps3eye-20221203.zip](https://github.com/opentrack/opentrack/files/10145451/ps3eye-20221203.zip)

**Reply by sopryshko** [reply]
Thank you. Now it works better, but the problem still occurs every 40-60 minutes. I also noticed that the FPS rate became more stable, which is good

**Reply by sthalik** [reply]
A crappy workaround would be to add use the restart tracker keyboard shortcut.

You could try setting `ps3eye-subprocess.exe` to realtime priority before starting the game. If it doesn't help it's probably the USB controller that can't handle the framerate. You could try plugging the camera to the other controller (one has black connectors and the other one blue ones)..

**Reply by sopryshko** [reply]
Hi. After two months I still have the same issue, during intensive dogfights it makes the game unplayable. I've set Opentarck and TrackIR to real-time priority using Task Manager, but I haven't found `ps3eye-subprocess.exe`. I tried to use different USB ports and hubs. I also tried to change settings (framerate, resolution, LED diameter etc.). Here is my config file: [config.txt](https://github.com/opentrack/opentrack/files/10517701/config.txt)

**Reply by sthalik** [reply]
You're using the CL-Eye driver based on the camera name. FWIW, here are the instructions using the libusb driver: https://github.com/opentrack/opentrack/wiki/PS3-Eye-open-driver-instructions.

**Reply by sopryshko** [reply]
Wow, thank you very much, that fixed my problem! Probably, the old driver that I had been using for years was broken after a regular Windows update.

**Comment by sopryshko** [comment]
Hello again) Is it possible to build the Open driver for arm64 systems? I can't install it on my arm64-based laptop which I got recently...

**Comment by JachVina** [comment]
Hello guys. Is there a way to have the MACOSX files a dmg? For some reason the make-app-bundle.sh files is not executing. Ive ran it in terminal and gets a "process completed''. Can someone help?
